presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/CachingHiveMetastore.java

```diff
@@ -170,21 +174,75 @@ public PartitionStatistics load(HivePartitionName key)
         partitionFilterCache = newCacheBuilder(expiresAfterWriteMillis, refreshMills, maximumSize)
                 .build(asyncReloading(CacheLoader.from(this::loadPartitionNamesByFilter), executor));
 
-        partitionCache = newCacheBuilder(expiresAfterWriteMillis, refreshMills, maximumSize)
-                .build(asyncReloading(new CacheLoader<HivePartitionName, Optional<Partition>>()
-                {
-                    @Override
-                    public Optional<Partition> load(HivePartitionName partitionName)
-                    {
-                        return loadPartitionByName(partitionName);
-                    }
+        if (isPartitioningEnabled) {
+            this.isPartitionVersioningEnabled = true;
```

Comment:
I'm a bit confused, why do we set this to `true` here? If `isPartitioningEnabled` is renamed per the above comment, I think we could just say `this.partitionVersioningEnabled = partitionVersioningEnabled` outside of this if-block?